cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -316,15 +325,15 @@ impl YaccParser {
                 i = self.parse_ws(k, false)?;
                 let num_newlines = self.num_newlines;
                 while i < self.src.len() && num_newlines == self.num_newlines {
-                    let (j, n, _) = self.parse_token(i)?;
-                    if self.ast.precs.contains_key(&n) {
-                        return Err(self.mk_error(YaccParserErrorKind::DuplicatePrecedence, i));
+                    let (j, n, span) = self.parse_token(i)?;
+                    if let Some((_, orig_span)) = self.ast.precs.get(&n) {
+                        self.duplicate_precedence_spans.entry(*orig_span).or_insert_with(Vec::new).push(span);
                     }
                     let prec = Precedence {
                         level: prec_level,
                         kind,
                     };
-                    self.ast.precs.insert(n, prec);
+                    self.ast.precs.insert(n, (prec, span));
```

Comment:
Now that we don't return early, this needs to be and else/match and probably use entry.